### PR TITLE
Typings for the Number Insight API

### LINKED ISSUE #362: Number Insight Typings
Create types for Number Insight API.

### PR DESCRIPTION
**Pull request by sudiptog81**
Adds typings for the Number Insight API.

## Description
Added typings as per the [API Reference](https://developer.nexmo.com/api/number-insight) and [code snippets](https://developer.nexmo.com/number-insight/code-snippets/). Prettier has also been added as a development dependency as the files in `src` had different styles than the existing typings in `typings/index.d.ts`.

## Motivation and Context
Shall contribute to better developer experience of the SDK in TypeScript projects. Done as part of Hacktoberfest 2020. Closes #362.

## Testing Details
Tests not affected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by kellyjandrews** [comment]
Thank you for your submission - we will get this reviewed and merged as soon as we can!

**Comment by kellyjandrews** [comment]
Hey @sudiptog81 - these changes are great, however there are two distinct PRs here, one for the prettier update, and the other is the number insights.  Please revert the changes for prettier, and open a separate PR for that change.

**Comment by sudiptog81** [comment]
I am reverting changes to prettier and package.json in the next commit 👍

**Comment by sudiptog81** [comment]
I resolved the merge conflicts and removed prettier changes. Let me know if there are more suggestions! Happy to address. There were some inconsistencies with the indentation, hence thought a prettier configuration could be useful.

File: typings/index.d.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
  *       CheckResponse -> VerifyCheckResponse
  */
 
-declare module '@vonage/server-sdk' {
```
Comment (kellyjandrews) [inline typings/index.d.ts]: Looks like this commit needs to happen after the prettify changes, as we are back into the formatting changes from the original. If you could remove the prettify changes from the file, then submit that way, then we can see this comparison in the prettify PR.  Sorry for the hassle!

**Review by kellyjandrews — CHANGES_REQUESTED** [review]

File: typings/index.d.ts (hunk 1 above)
Comment (kellyjandrews) [reply typings/index.d.ts]: What I'm looking to isolate is the formatting into one PR, and the actual number insight addition, in a different one. Does that make sense?

Comment (sudiptog81) [reply typings/index.d.ts]: Sure! Understood the issue, one way I can handle this is by replacing the contents of this file with the actual file on `master` and then I include the Number Insight typings. Works?

File: typings/index.d.ts [hunk 2]
```diff
@@ -341,6 +341,120 @@ declare module '@vonage/server-sdk' {
...
     }
 
+    /* number-insight API */
+    export interface NumberInsightError extends VonageApiError {
+        status: NumberInsightResponseStatusCode | number;
+        error_text: string;
+        [key: string]: any;
+    }
+
+    export interface NumberInsightObject {
```
Comment (kellyjandrews) [inline typings/index.d.ts:351]: Looks like you covered some of the options, but a few are still missing - was that intentional?  https://github.com/Vonage/vonage-node-sdk/blob/master/src/NumberInsight.js#L30-L75

Comment (sudiptog81) [reply typings/index.d.ts:351]: Completely overlooked it! Apologizing for this, I was focusing more on `NumberInsightResponse` it seems.

Comment (sudiptog81) [reply typings/index.d.ts:351]: Added the missing options: `features`, `callback`, `callback_timeout`, `callback_method`, `client_ref`, `'include-intermediate-callbacks'`.

**Review by kellyjandrews — APPROVED** [review]
These look good to me - I'll get them merged in!

**Pull request merged**